### ISSUE TITLE
Background corruption when changing screen layout

### ISSUE THREAD

**Issue by phocean** [mate-desktop/mate-session-manager#159]
#### Expected behaviour

The background geometry should get automatically adapted to any screen layout change (number of external screens, of various resolutions and orientations).

#### Actual behaviour

When Mate is responsible of handling the desktop drawing, some corruption may happen.

Here is the wallapaper setting:

![capture du 2018-05-07 05-05-59](https://user-images.githubusercontent.com/4280885/39684163-22b042f4-51ba-11e8-95b1-7c482b4151b4.png)

Here is what happens when an external screen is plugged in or out:

![capture du 2018-05-07 05-03-31](https://user-images.githubusercontent.com/4280885/39684177-37862f90-51ba-11e8-89a6-622e92830ecc.png)

![capture du 2018-05-07 05-05-35](https://user-images.githubusercontent.com/4280885/39684179-3b49bb2e-51ba-11e8-9125-16f1b95d18bc.png)

![capture du 2018-05-07 05-03-10](https://user-images.githubusercontent.com/4280885/39684187-45c02584-51ba-11e8-8f06-fde6bd593052.png)

#### Steps to reproduce the behavior

1. Disable Caja handling wallpaper ("don't show desktop icons setting") so that it is handled by mate-session

2. Plug in an external monitor and observe wallpaper corruption (generally on the external screen)

3. Plug out the external monitor and observe how corruption is still present on primary screen (it is not possible to recover a clean state)

4. A way to recover a clean wallpaper geometry is to simply open the _background_ tab in _Appearance_.

#### Possible workaround

This has worked so far for me, even though I do not understand what it really does behind.

Using _dconf_, in `/org/mage/desktop/background`, disabling `draw-background` makes the issue vanish.

#### MATE general version

1.20.1 as in Ubuntu 18.04.

**Comment by raveit65** [comment]
What kind of monitors are you using?
Was this working before with older MATE versions?

**Comment by phocean** [comment]
They are connected through DP and HDMI ports.

Any combination (one or two external monitor on any interface) of any screen model (it also happen at my work place with different screen models and resolutions) would cause some corruption.

If it helps, here is the `xrandr` output with the full layout:

[xrandr.txt](https://github.com/mate-desktop/mate-session-manager/files/1982648/xrandr.txt)

Screens are DELL U2515H and EIZO EV2450.

**Comment by phocean** [comment]
And sorry, I cannot reply on previous versions as I am a new user :-)

But testing with an old live ISO is definitely something I could do if it helps.

**Comment by raveit65** [comment]
In other words it is a fresh installation of  ubuntu-18.04 ?
And you are using non-HIDPI resolutions and HIDPI resolutions?
@flexiondotorg 
Please help out, this is ubuntu distro...

**Comment by phocean** [comment]
Yes, it is a fresh install of Ubuntu 18.04, so I would be surprised if it cannot be reproduced on any setup with a dual screen.

At office none of my screen is HDPI (1920x1080). And here one of them is not (also 1920x1080).

**Comment by phocean** [comment]
No, it is my laptop moving between places, so same system, same issue.

**Comment by phocean** [comment]
The gsettings trick does not work for me, so we may not have the same issue.

You tested with "show-desktop-icons" disabled, right ? (so that caja is not taking care of the drawing).

The best workaround for me is to actually rely on caja, "show-desktop-icons" enabled.

**Comment by phocean** [comment]
Two new computers : Lenovo Thinkpad T440p (Nvidia graphics), Lenovo Yoga S730.

Fresh installs, same issue.

![Capture du 2019-04-24 20-01-31](https://user-images.githubusercontent.com/4280885/56682666-5829b280-66cc-11e9-981f-f6c877617ee5.png)
